app/helpers/checks.py

```diff
@@ -0,0 +1,89 @@
+from urllib.parse import urlparse
+
+from boto3.dynamodb.conditions import Key
+from flask import abort
+
+from app.helpers.response_generation import make_error_msg
+from app import app
+
+config = app.config
+
+
+def check_params(scheme, host, url, base_path):
+    """
+    * Quick summary of the function *
+
+    In the process to create a shortened url, this is the first step, checking all parameters. If
+    they all fall within expectation, this function return the 'base url' that will lead to the
+    redirection link. On production, that base_url would be http(s)://s.geo.admin.ch/.
+
+    * Abortions originating in this function *
+
+    Abort with a 400 status code if there is no url, given to shorten
+    Abort with a 400 status code if the application hostname can't be determined
+    Abort with a 400 status code if the application domain is not one of the allowed ones.
+
+    * Abortions originating in functions called from this function *
+
+    None
+
+    * parameters and return values *
+
+    :param scheme: the scheme (http | https) used to make the query
+    :param host: the hostname of this service
+    :param url: the url given to be shortened
+    :param base_path: the reverse proxy paths in front of this service
+    :return: the base url to reach the redirected url.
+    """
+    if url is None:
+        abort(make_error_msg(400, 'url parameter missing from request'))
+    hostname = urlparse(url).hostname
+    if hostname is None:
+        abort(make_error_msg(400, 'Could not determine the query hostname'))
+    domain = ".".join(hostname.split(".")[-2:])
+    if domain not in config['allowed_domains'] and hostname not in config['allowed_hosts']:
+        abort(make_error_msg(400, f'Service shortlink can only be used for '
```

Comment:
the **and** is indeed correct. If either the hostname or the domain of the url given to be shortened is allowed,the url is valid to be shortened.

The malicious use possible with this information would be to perform DOS attacks to forge acceptable queries which will redirect to invalid addresses on our services, filling the dynamodb with useless data. It might make sense not to send the information back (especially since it's an internal tool).